src/main/java/com/firebolt/jdbc/type/lob/FireboltBlob.java

```diff
@@ -0,0 +1,160 @@
+package com.firebolt.jdbc.type.lob;
+
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.sql.Blob;
+import java.sql.SQLException;
+import java.util.Arrays;
+import java.util.LinkedList;
+import java.util.List;
+
+public class FireboltBlob implements Blob {
+    private byte[] buf = new byte[0];
+
+    public FireboltBlob() {
+        this(new byte[0]);
+    }
+
+    public FireboltBlob(byte[] buf) {
+        this.buf = buf;
+    }
+
+    @Override
+    public long length() throws SQLException {
+        return buf.length;
+    }
+
+    @Override
+    public byte[] getBytes(long pos, int length) throws SQLException {
+        isValid();
+        int from = (int)pos - 1;
+        if (from < 0 || from > length()) {
+            throw new SQLException("Invalid position in Clob object set");
+        }
+        if (from + length > length()) {
+            throw new SQLException("Invalid position and substring length");
+        }
+        byte[] bytes = new byte[length - from];
```

Comment:
No, this is exactly the purpose of this parameter. Variable `from` is calculated as `pos - 1` because pos is 1-based and is defined as following in javadoc:
```
pos – the ordinal position of the first byte in the BLOB value to be extracted; the first byte is at position 1
```

Here is the full javadoc for your convenience:

```
Retrieves all or part of the BLOB value that this Blob object represents, as an array of bytes. This byte array contains up to length consecutive bytes starting at position pos.
Params:
pos – the ordinal position of the first byte in the BLOB value to be extracted; the first byte is at position 1 length – the number of consecutive bytes to be copied; the value for length must be 0 or greater
Returns:
a byte array containing up to length consecutive bytes from the BLOB value designated by this Blob object, starting with the byte at position pos
```

Reply:
But why do we always substract length - from. What if we have 
`1, 2, 3, 4, 5`
and `from=2, length=2`
We should return  `3, 4`, and we'll return an empty array since we allocate 0 memory (`length - from`)

Reply:
fixed